File: wrappers/dotnet/indy-sdk-dotnet-test/AnonCredsTests/ProverCreateMasterSecretTest.cs
using Hyperledger.Indy.AnonCredsApi;
using Hyperledger.Indy.Test.Util;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.AnonCredsTests
{
    [TestClass]
    public class ProverCreateMasterSecretTest : AnonCredsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestProverCreateMasterSecretWorks()
        {
        }

        [TestMethod] 
        public async Task TestProverCreateMasterSecretWorksForDuplicate()
        {
            var ex = await Assert.ThrowsExceptionAsync<DuplicateMasterSecretNameException>(() =>
               AnonCreds.ProverCreateMasterSecretAsync(wallet, masterSecretId)
           );
        }

        [TestMethod]
        public async Task TestProverCreateMasterSecretWorksForEmptyName()
        {
            await AnonCreds.ProverCreateMasterSecretAsync(wallet, null);
        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/anoncreds/ProverCreateMasterSecretTest.java
package org.hyperledger.indy.sdk.anoncreds;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.isA;

import java.util.concurrent.ExecutionException;

public class ProverCreateMasterSecretTest extends AnoncredsIntegrationTest {

	@Test
	public void testProverCreateMasterSecretWorks() throws Exception {
	}

	@Test
	public void testProverCreateMasterSecretWorksForDuplicate() throws Exception {

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(DuplicateMasterSecretNameException.class));

		Anoncreds.proverCreateMasterSecret(wallet, masterSecretId).get();
	}
}
